pkg/metrics/sampler/matcher.go

```diff
@@ -0,0 +1,209 @@
+//  Copyright 2020 New Relic Corporation. All rights reserved.
+//  SPDX-License-Identifier: Apache-2.0
+
+package sampler
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/newrelic/infrastructure-agent/pkg/log"
+)
+
+var (
+	logger          = log.WithComponent("sampler-matcher")
+	typesToEvaluate = map[string]bool{"ProcessSample": true}
+)
+
+// ExpressionMatcher is an interface every evaluator must implement
+type ExpressionMatcher interface {
+	// Evaluate evaluates a property
+	Evaluate(event interface{}) bool
+}
+
+type attributeCache map[string]string
+
+var attrCache attributeCache
+
+type regexCompiledCache map[string]*regexp.Regexp
+
+var regexCache regexCompiledCache
+
+func init() {
+	attrCache = attributeCache{}
+	attrCache = make(map[string]string)
+	attrCache["process.name"] = "ProcessDisplayName"
+	attrCache["process.executable"] = "CmdLine"
+
+	regexCache = regexCompiledCache{}
+}
+
+type matcher struct {
+	PropertyName  string
+	ExpectedValue interface{}
+	Evaluator     func(expected interface{}, actual interface{}) bool
+}
+
+func (p matcher) Evaluate(event interface{}) bool {
```

Comment:
could we add some unit testing to this Evaluate logic?

Reply:
It is covered